docs/courses/skill.md

```diff
@@ -0,0 +1,173 @@
+- `Skill > Thumbnails`: A list of filenames of the thumbnails to be used on the course page to give an idea of the skill. A list of available files can be found on [`apps/web/static/images/`](https://github.com/kantord/LibreLingo/tree/main/apps/web/static/images). The names should be used without extension and without `_tiny` or `_tinier` parts. e.g. `banana2_tinier.jpg` should be written as `banana2`.
+
+**`New words`** has a list of new words that the lesson asks about.
+- `Word`: The word in destination language.
+- `Synonyms`: A list of synonyms of the above word. (optional)
+- `Translation`: Translation of the word in target audience's language.
+- `Also accepted`: A list of synonyms of the `Translation` word above. (optional)
+- `Images`: A list of images for the word that is defined under `Word` or `Translation` above. The image names has to be without extension and without `_tiny` or `_tinier` part.
+
+**`Phrases`** has a list of sentences or phrases that the lession asks about.
+- `Phrase`: The phrase in destination language. e.g. In this case, it's written in Spanish.
+- `Alternative versions`: An alternative version of the above phrase. (optional)
+- `Translation`: Translation of the phrase in target audience's language. e.g. In this case, it's written in English.
+
+**`Mini-dictionary`** has a list of words and meanings. [TODO: Explain where and how it is used]
+- `<destination language>`: A list of words in destination language as key and meaning in target audience's language.
+- `<target audience's language>`: A list of words in target audience's language as key and meaning in destination language.
+
+<a id="creating-new"></a>
+## Creating new skills
+
+Creating new skills is a matter of creating their corresponding YAML file.
+That being said, there are a couple things to keep in mind:
+
+- Skill yaml file should be inside a [module](module.md)'s `skills` directory.
+- Skills should have an unique `Id`. Such identifiers can be generated using
+  [this website](https://www.uuidgenerator.net/version4).
+- Before skills will actually show up in the web app, they need to be added to
+  a [`module.yaml`](module.md#yaml).
+
+<a id="example-edit"></a>
+## Examples of editing a skill
+
+For example, this is how you'd want to add a new alternative version to the
+phrase "La femme dit bonjour":
+
+```diff
+Phrases:
+  - Phrase: La femme dit bonjour
+    Alternative versions:
+      - La femme dit salut
++     - La madame dit salut
+    Translation: The woman says hello
+    Alternative translations:
+      - The woman says hi
+```
+
+Keep in mind that when you add a new phrase, or a new version or translation
+to a phrase, it's not automatically updating to the mini-dictionary. So you'd
+want to add any new words there:
```

Comment:
```suggestion
add any new words there:
```

Reply:
Also maybe we should mention that you need to add the new words in both languages

Reply:
I've added something like this. Resolve this if it's acceptable.
````
Also note, that the same phrase should be added in both the languages under `Mini-dictionary` key. Meaning, if you add a phrase under target language, you should add it under the source language too. For example:

```yaml
Mini-dictionary:
  French:
    - dit: says
    - bonjour:
      - hello
      - hi
...
  English:
    - says: dit
    - hello:
      - bonjour
      - salut
```
````

Reply:
> Also note, that the same phrase should be added in both the languages under `Mini-dictionary` key. Meaning, if you add a phrase under target language, you should add it under the source language too. For example:

Yes, but I would maybe simplify it:

> Also note, that the new words from each phrase need to be added for both the source and the target language in the `Mini-dictionary`.  For example:

Reply:
I think a clarification would be helpful to users.

> new words from each phrase

I have these questions appear to me when I read this:
- What does it mean by "new words"?
- New words from phrases from where? Is this referring to phrases under `Phrases` key?

Reply:
> is this referring to phrases under Phrases key?

yes

> What does it mean by "new words"?

Words that haven't been defined earlier. The system can re-use definitions from other Skills, although duplicates are also allowed, so it's not an issue if the same definition is included twice.

Reply:
Maybe something like this then...?
```
Also note, that the new words from each phrase listed under `Phrases` key need to be added for both the source and the target language in the `Mini-dictionary`. For example:

...(example code)...

Don't worry about listing the same word more than once in a course. It will not cause any issues if you do so.
```